src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeColorization.cs

```diff
@@ -0,0 +1,244 @@
+using Standard;
+using System.Windows.Appearance;
+using System.Windows.Media;
+using Microsoft.Win32;
+using System.Collections.Generic;
+
+namespace System.Windows;
+
+internal static class ThemeColorization
+{
+    // ------------------------------------------------
+    //
+    // Members
+    //
+    // ------------------------------------------------
+    #region Private Members
+    /// <summary>
+    /// Registry Path containing current theme information.
+    /// </summary>
+    private static readonly string _regThemeKey = "HKEY_CURRENT_USER\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Themes";
+
+    private static readonly string _appThemeKey = "HKEY_CURRENT_USER\\Software\\Microsoft\\Windows\\CurrentVersion\\Themes\\Personalize";
+
+    /// <summary>
+    /// Static member indicating theme which is currently applied.
+    /// </summary>
+    private static string _currentApplicationTheme;
+
+    private static string _currentAppsTheme = "Light";
+
+    private static bool _isFluentWindowsThemeEnabled = false;
+
+    private static readonly string _themeDictionaryName = "FluentWindows.xaml";
+
+    private static List<Uri> _themeDictionaryUris = new List<Uri>();
+
+    #endregion
+
+    #region Constructor
+
+    static ThemeColorization()
+    {
+        foreach (ResourceDictionary mergedDictionary in Application.Current.Resources.MergedDictionaries)
```

Comment:
Will crash when there is no application and won't work for window level theming.

Reply:
Added a check for Application.Current before enabling this flag.